scipy/stats/stats.py

```diff
@@ -1888,24 +1894,71 @@ def scoreatpercentile(a, per, limit=(), interpolation_method='fraction',
     49.5
 
     """
-    # adapted from NumPy's percentile function.  When we require numpy >= 1.8,
-    # the implementation of this function can be replaced by np.percentile.
     a = np.asarray(a)
+
+    if np.ndim(per) > 1:
+        raise ValueError("`per` must be 1d.")
+
+    # patch for numpy/numpy#18158. nan
+    # If zero size, only returned shape matters.
     if a.size == 0:
-        # empty array, return nan(s) with shape matching `per`
-        if np.isscalar(per):
-            return np.nan
-        else:
-            return np.full(np.asarray(per).shape, np.nan, dtype=np.float64)
+        with warnings.catch_warnings() as _, \
```

Comment:
I believe we tend to avoid the `\` line continuations in favor of bounding parentheses

Reply:
We/I do, but the natural thing:
```python3
with (warnings.catch_warnings() as _, 
      np.errstate(invalid='ignore') as _):
```
I did [a little research](https://www.google.com/search?q=parentheses+around+context+managers+python) and didn't find a workaround that's any prettier than this.